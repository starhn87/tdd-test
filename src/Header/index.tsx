import React from "react";
import { useLocation } from "react-router-dom";

const Header = () => {
  const { pathname } = useLocation();
  let title = "에러";

  if (pathname === "/") {
    title = "할 일 목록";
  } else if (pathname === "/add") {
    title = "할 일 추가";
  } else if (pathname.startsWith("/detail")) {
    title = "할 일 상세";
  }

  return (
    <div>
      <div>{title}</div>
    </div>
  );
};

export default Header;
