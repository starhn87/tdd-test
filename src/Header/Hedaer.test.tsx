import React from "react";
import { render, screen } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import Header from ".";

describe("<Header />", () => {
  it("renders component correctly", () => {
    const history = createMemoryHistory();
    history.push("/");

    const { container } = render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );

    const label = screen.getByText("할 일 목록");
    expect(label).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("renders component correctly with /add URL", () => {
    const history = createMemoryHistory();
    history.push("/add");

    const { container } = render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );

    const label = screen.getByText("할 일 추가");
    expect(label).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("renders component correctly with /detail/:id URL", () => {
    const history = createMemoryHistory();
    history.push("/detail/1");

    render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );

    const label = screen.getByText("할 일 상세");
    expect(label).toBeInTheDocument();
  });

  it("renders component correctly with NotFound", () => {
    const history = createMemoryHistory();
    history.push("/not_found");

    render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );

    const label = screen.getByText("에러");
    expect(label).toBeInTheDocument();
  });
});
