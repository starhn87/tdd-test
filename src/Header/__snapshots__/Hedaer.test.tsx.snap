// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders component correctly 1`] = `
<div>
  <div>
    <div>
      할 일 목록
    </div>
  </div>
</div>
`;

exports[`<Header /> renders component correctly with /add URL 1`] = `
<div>
  <div>
    <div>
      할 일 추가
    </div>
  </div>
</div>
`;
